dev/ProgressBar/ProgressBar.cpp

```diff
@@ -0,0 +1,154 @@
+{
+    // TODO: things
+}
+
+void ProgressBar::OnLayoutRootLoaded(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    // TODO: things
+}
+
+void ProgressBar::OnRangeBaseValueChanged(const winrt::DependencyObject& sender, const winrt::DependencyProperty& args)
+{
+    // NOTE: This hits when the Value property changes, because we called RegisterPropertyChangedCallback.
+
+    SetProgressBarIndicatorWidth();
+}
+
+void ProgressBar::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    // NOTE: This hits when IsIndeterminate changes because we set MUX_PROPERTY_CHANGED_CALLBACK to true in the idl.
+
+    UpdateStates();
+}
+
+void ProgressBar::OnShowPausedPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressBar::OnShowErrorPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressBar::UpdateStates()
+{
+    if (ShowError())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Error", true);
+    }
+    else if (ShowPaused() && IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Error", true); // Paused-Indeterminate state same visual treatment as Error state
+    }
+    else if (ShowPaused())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Paused", true);
+    }
+    else if (IsIndeterminate())
+    {
+        UpdateWidthBasedTemplateSettings();
+        winrt::VisualStateManager::GoToState(*this, L"Indeterminate", true);
+    }
+    else if (!IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Determinate", true);
+    }
+}
+
+void ProgressBar::SetProgressBarIndicatorWidth()
+{
+    if (auto progressBarIndicator = m_progressBarIndicator.get())
+    {
+        double progressBarWidth = m_layoutRoot.get().ActualWidth();
+        double maximum = Maximum();
+        double minimum = Minimum();
+        double increment = 0;
+        
+        increment = progressBarWidth / (maximum - minimum);
+        progressBarIndicator.Width(increment * (Value() - minimum));
+    }
+}
+
+void ProgressBar::UpdateWidthBasedTemplateSettings()
+{
+    auto const templateSettings = winrt::get_self<::ProgressBarTemplateSettings>(TemplateSettings());
+    auto progressBar = m_layoutRoot.get();
+    auto progressBarIndicator = m_progressBarIndicator.get();
+
+    auto const [width, height] = [progressBar]()
+    {
+        if (progressBar)
+        {
+            float const width = static_cast<float>(progressBar.ActualWidth());
+            float const height = static_cast<float>(progressBar.ActualHeight());
+            return std::make_tuple(width, height);
+        }
+        return std::make_tuple(0.0f, 0.0f);
+    }();
+
+    progressBarIndicator.Width(width / 3);
+
+    templateSettings->ContainerAnimationEndPosition(width);
+    templateSettings->ContainerAnimationStartPosition(0);
```

Comment:
Template settings are APIs too, since this value never changes the api probably shouldn't exist. It does add a bit of customization and for completeness maybe it should be included but I think we should remove this template setting.

Reply:
@jevansaks  This exists on the framework version of TemplateSettings, if we want this control to be a drop-in replacement for it, do we need to keep all the old Template Setting properties. This property was only used to offset the ellipses. If we do need this, does that mean we need all the other ellipse properties?

EllipseAnimationEndProperty
EllipseAnimationWellPosition
EllipseDiameter
EllipseOffset

Reply:
Were they used in the old ProgressBar template? The only reason to keep them would be to maintain compatibility with someone who was using the *old* ProgressBar template with the *new* control.

Reply:
@YuliKl Should these Template Setting properties be kept?

Reply:
Maybe they should be kept if they are useful for any dev that may need to re-template the control in the future,

Reply:
> The only reason to keep them would be to maintain compatibility with someone who was using the _old_ ProgressBar template with the _new_ control.

I don't think we need to support this scenario. 

In general, my preference is to streamline the API. We can always add features, like TemplateSettings, later, right? And I definitely don't want us adding back the Ellipse* properties  - we're getting rid of the dots, so these properties should no longer apply.